Broadcast/Helpers/TwitterClientManager.swift

```diff
@@ -0,0 +1,331 @@
+    
+    let copy = draft
+    DispatchQueue.global(qos: .default).async {
+      let newDraft = Draft.init(context: self.draftsStore.context)
+      newDraft.date = Date()
+      newDraft.text = copy.text
+      // TODO: Fix draft media
+      //newDraft.media = copy.media?.fixedOrientation.pngData()
+      newDraft.id = UUID()
+      
+      self.draftsStore.save()
+    }
+    
+    withAnimation {
+      self.draft = .init()
+      self.state = .idle
+    }
+  }
+  
+  /// Retrieve the specified draft from CoreData, storing it in memory and deleting it from the CoreData database
+  /// - Parameter draft: The chosen draft for retrieval and deletion
+  func retreiveDraft(draft: Draft) {
+    withAnimation {
+      self.draft = MutableTweet(text: draft.text)
+      
+      // TODO: Fix draft media
+      //      if let media = draft.media {
+      //        self.draft.media = UIImage(data: media)
+      //      }
+    }
+    
+    let managedObjectContext = PersistanceController.shared.context
+    managedObjectContext.delete(draft)
+    
+    PersistanceController.shared.save()
+  }
+}
+
+// MARK: Models
+extension TwitterClientManager {
+  enum State: Equatable {
+    case idle, busy, initializing
+    case error(_: String? = nil)
+    
+    static var genericTextError = State.error("Oh man, something went wrong sending that tweet. It might be too long.")
+    static var genericTextAndMediaError = State.error("Oh man, something went wrong sending that tweet. Maybe it’s too long, or your chosen media is causing a problem.")
+  }
+  
+  struct ClientCredentials {
+    static private var plist: NSDictionary? {
+      guard let filePath = Bundle.main.path(forResource: "TwitterAPI-Info", ofType: "plist") else {
+        fatalError("Couldn't find file 'TwitterAPI-Info.plist'.")
+      }
+      // 2
+      return NSDictionary(contentsOfFile: filePath)
+    }
+    
+    static var apiKey: String {
+      guard let value = plist?.object(forKey: "API_KEY") as? String else {
+        fatalError("Couldn't find key 'API_KEY' in 'TwitterAPI-Info.plist'.")
+      }
+      
+      return value
+    }
+    
+    static var apiSecret: String {
+      guard let value = plist?.object(forKey: "API_SECRET") as? String else {
+        fatalError("Couldn't find key 'API_KEY' in 'TwitterAPI-Info.plist'.")
+      }
+      
+      return value
+    }
+    
+    static var credentials: OAuthCredentials {
+      .init(key: apiKey, secret: apiSecret)
+    }
+    
+    static var __authQueryString: String {
+      guard let value = plist?.object(forKey: "__TEST_AUTH_QUERY_STRING") as? String else {
+        fatalError("Couldn't find key '__TEST_AUTH_QUERY_STRING' in 'TwitterAPI-Info.plist'.")
+      }
+      
+      return value
+    }
+    
+    static var callbackProtocol = "twitter-broadcast://"
+    static var callbackURL: URL {
+      URL(string: callbackProtocol)!
+    }
+  }
+}
+
+struct TypeaheadResponse: Decodable {
+  var num_results: Int
+  var users: [User]?
+}
```

Comment:
This should be updated with a v1 user definition (or extend `Twift.User` to allow decoding from a v1 user):

```swift
struct V1User: Codable {
  let id_str: String
  let name: String
  let screen_name: String
  let profile_image_url_https: URL

  func toUser() -> User {
    let jsonString = """{
id: "\(id_str)",
profileImageUrl: "\(profile_image_url_https)",
name: "\(name)",
userName: "\(screen_name)"
}"""

    return try! JSONDecoder().decode(User.self, from: jsonString.data(using: .utf8)!)
  }
}
```